tox.ini

```diff
@@ -0,0 +1,39 @@
+[tox]
+isolated_build = True
+envlist =
+    pre-commit
+    check-licenses
+    py{38,39,310,311}
+
+[gh-actions]
+python =
+    3.8: py38
+    3.9: py39
+    3.10: pre-commit,check-licenses, py310
+    3.11: py311
+
+[testenv:pre-commit]
+deps=pre-commit
+commands=pre-commit run --all-files --show-diff-on-failure
+
+[testenv:check-licenses]
+deps=
+    liccheck==0.7.3
```

Comment:
can we have soft dependency versions here like we had in `pyproject.toml`?  (same for pytest, and pytest-cov)

Reply:
Good point. Updated.